metrics/internal/collectors/rbd-mirror.go

```diff
@@ -163,29 +164,40 @@ func (c *RBDMirrorCollector) Collect(ch chan<- prometheus.Metric) {
 					default:
 						klog.Errorf("Unknown state %q of image %q", site.State, image.Name)
 					}
-
-					desc := cache.RBDMirrorPeerSiteDescription{}
-					err := json.Unmarshal([]byte(description), &desc)
-					if err != nil {
-						klog.Errorf("Failed to unmarshal description of image %q from site %q: %v", image.Name, site.SiteName, err)
-						return
+				}
+				if image.Description == "local image is primary" {
+					// site.Description can have values like "replaying,{...}" "split-brain" etc.
+					siteDescription := strings.SplitN(site.Description, ", ", 2)
+					if len(siteDescription) != 2 {
+						klog.Errorf("Unexpected mirror peer site description %q of image %q to site %q.", site.Description, image.Name, site.SiteName)
```

Comment:
in this case, we can continue? we can remove else check if we continue here

Reply:
fixed.